omnipaxos_core/src/storage.rs

```diff
@@ -233,27 +239,27 @@ impl<T> StateCache<T>
     fn set_compacted_idx(&mut self, compacted_idx: u64) {
         self.compacted_idx = compacted_idx;
     }
+    fn set_accepted_idx(&mut self, acc_idx: u64) {
+        self.acc_idx = acc_idx;
+    }
 
     // Appends an entry to the end of the `batched_entries`. If the batch is full, the
     // batch is flushed and return flushed entries. Else, return None.
     fn append_entry(&mut self, entry: T) -> Option<Vec<T>> {
         self.batched_entries.push(entry);
-        self.flush_if_batch_full()
+        self.pop_batched_entries_tobe_flushed()
     }
     // Appends entries to the end of the `batched_entries`. If the batch is full, the
     // batch is flushed and return flushed entries. Else, return None.
     fn append_entries(&mut self, entries: Vec<T>) -> Option<Vec<T>> {
         self.batched_entries.extend(entries);
-        self.flush_if_batch_full()
+        self.pop_batched_entries_tobe_flushed()
     }
-    // Flushes the batched entries if the batch is full, and returns the flushed entries.
-    fn flush_if_batch_full(&mut self) -> Option<Vec<T>> {
-        let mut flushed_entries = Vec::with_capacity((self.batched_entries.len() / self.batch_size) * self.batch_size);
-        while self.batched_entries.len() >= self.batch_size {
-            flushed_entries.extend(self.batched_entries.drain(0..self.batch_size).collect::<Vec<T>>());
-        }
-        if !flushed_entries.is_empty(){
-            Some(flushed_entries)
+    // Return batched entries if the batch is full that need to be flushed in to storage.
+    fn pop_batched_entries_tobe_flushed(&mut self) -> Option<Vec<T>> {
+        let flushed_entries_len = (self.batched_entries.len() / self.batch_size) * self.batch_size;
+        if flushed_entries_len > 0 {
```

Comment:
Is this calculation really needed? Isn't `flushed_entries_len` always the self.batch_size?

Reply:
If `batched_entries` == 35 and `batch_size` == 10, the `flushed_entries_len` will be 30 (not 10).

Reply:
Why would we want to flush 30 entries in this case? To me, it makes more sense to either flush all 35 entries or the 10 entries that the user set as batch size.

Reply:
For simplicity, if we go over the batch size, then let's just flush all entries.